sklearn/tests/test_calibration.py

```diff
@@ -1077,3 +1063,21 @@ def test_calibrated_classifier_deprecation_base_estimator(data):
     warn_msg = "`base_estimator` was renamed to `estimator`"
     with pytest.warns(FutureWarning, match=warn_msg):
         calibrated_classifier.fit(*data)
+
+
+def test_calibration_groupkfold(data):
+    # Check that groups are routed to the splitter
+    X, y = data
+    split_groups = np.array([0, 1] * (len(y) // 2))
+
+    class MyGroupKFold(GroupKFold):
+        """Custom Splitter that checks that the values of groups are correct"""
+
+        def split(self, X, y=None, groups=None):
+            assert (groups == split_groups).all()
+            return super().split(X, y=y, groups=groups)
```

Comment:
`GroupKFold` raises if `groups` is None anyway, you won't need this custom class to test.

Reply:
This was more about checking that the correct groups data is being passed. If you think it's not necessary, I'd rather remove it to simplify the test.

Reply:
I don't think we have to check for the correctness of the groups. Those mechanisms are tested elsewhere.